client-core/src/mnemonic.rs

```diff
@@ -36,6 +36,7 @@ impl<'de> Deserialize<'de> for Mnemonic {
 
 impl Mnemonic {
     /// Generate and returns mnemonic words
+    #[allow(clippy::new_without_default)]
```

Comment:
Instead of this, can you create a `Default` implementation which calls `new()`

Reply:
I think the reason to ignore this (perhaps a comment could be added) is that Default should be deterministic, but `new()` generates a random mnemonic